app/javascript/packs/modals/unpublishAllPosts.js

```diff
@@ -0,0 +1,90 @@
+/* eslint-disable no-restricted-globals */
+import { closeWindowModal, showWindowModal } from '@utilities/showModal';
+import { request } from '@utilities/http';
+
+const unpublishAllPosts = async (event) => {
+  event.preventDefault();
+  const { userId } = event.target.dataset;
+
+  try {
+    // const http_request = await request(endpoint, { method: 'POST' });
+    // const response = await http_request.json();
+    const response = await request(
+      `/admin/member_manager/users/${userId}/unpublish_all_articles`,
+      {
+        method: 'POST',
+        body: JSON.stringify({ id: userId }),
+        credentials: 'same-origin',
+      },
+    );
+
+    const outcome = await response.json();
+
+    top.addSnackbarItem({
+      message: outcome.message,
+      addCloseButton: true,
+    });
+  } catch (error) {
+    top.addSnackbarItem({
+      message: `Error: ${error}`,
+      addCloseButton: true,
+    });
+  }
+
+  closeWindowModal(window.parent.document);
+};
+
+const modalContents = new Map();
+
+// function getConfirmation(event) {
+//   const confirmation = confirm('Are you sure you want to unpublish all posts?');
+//   const endpoint = event.target.dataset.formAction;
+
+//   if (confirmation) {
+//     unpublishAllPosts({ event, endpoint });
+//   }
+// }
+
+/**
+ * Helper function to handle finding and caching modal content. Since our Preact modal helper works by duplicating HTML content,
+ * and our modals rely on IDs to label form controls, we remove the original hidden content from the DOM to avoid ID conflicts.
+ *
+ * @param {string} modalContentSelector The CSS selector used to identify the correct modal content
+ */
+function getModalContents(modalContentSelector) {
+  if (!modalContents.has(modalContentSelector)) {
+    const modalContentElement =
+      window.parent.document.querySelector(modalContentSelector);
+    const modalContent = modalContentElement.innerHTML;
+
+    modalContentElement.remove();
+    modalContents.set(modalContentSelector, modalContent);
+  }
+
+  return modalContents.get(modalContentSelector);
+}
```

Comment:
I think we don't need this - we use this pattern in various places because when you replicate form inputs across the "hidden" modal content and the visible modal content, it causes havoc with duplicate IDs (e.g. I have something like `<label for="my-input">`, but two things with ID `my-input` exist on the page).

Since we don't have any form inputs in this modal, I don't think we need to worry about it. Instead, down on line 85 you could directly have:

```
modalContent: window.parent.document.querySelector(modalContentSelector)
```

Reply:
ahhh... nice! love it!